docs/CONTRIBUTING.md

```diff
@@ -2,17 +2,43 @@
 
 ## Setting Up Development Environment
 
-We use [poetry](https://python-poetry.org/) as our package and dependency manager.
+We use [uv](https://docs.astral.sh/uv/) as our package and dependency manager.
 
-To create a virtual environment for development, run the following in your shell:
+### Installation
+
+First, install `uv` using the official method:
+
+**macOS and Linux**
+
+```sh
+curl -LsSf https://astral.sh/uv/install.sh | sh
+```
+
+**Windows**
+
+```powershell
+powershell -c "irm https://astral.sh/uv/install.ps1 | iex"
+```
+
+For more detailed instructions, refer to the [official uv setup guide](https://docs.astral.sh/uv/#getting-started).
+
+### Create a Virtual Environment
+
+After installing `uv`, create a virtual environment for development by running:
```

Comment:
should we mention installing and using python 3.10 since that's the lowest version we support?

the `uv` docs cover this https://docs.astral.sh/uv/guides/install-python/#installing-a-specific-version

Reply:
@willbakst 
We can define default python version. I will add 3.10 in `.python-version`
> Project Python versions[#](https://docs.astral.sh/uv/concepts/python-versions/#project-python-versions)
By default uv python install will verify that a managed Python version is installed or install the latest version.

> However, a project may define a .python-version file specifying the default Python version to be used. If present, uv will install the Python version listed in the file.

Reply:
`.python-versions` is better than `.python-version`